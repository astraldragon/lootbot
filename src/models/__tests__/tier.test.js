const tier = require("../tier");

describe("models/tier", () => {
  beforeEach(() => {
    this.model = {
      hasMany: jest.fn("hasMany").mockReturnValue()
    };

    this.sequelize = {
      define: jest.fn("define").mockReturnValue(this.model)
    };

    this.dataTypes = {
      STRING: "string",
      INTEGER: "integer",
      FLOAT: "float"
    };

    this.models = {
      Loot: jest.fn("Tier"),
      Message: jest.fn("Message")
    };
  });

  it("should create the model definition", () => {
    tier(this.sequelize, this.dataTypes);
    expect(this.sequelize.define).toMatchSnapshot();
  });

  it("should associate tier with loot", () => {
    const Tier = tier(this.sequelize, this.dataTypes);
    Tier.associate(this.models);

    expect(Tier.hasMany).toHaveBeenCalledWith(this.models.Loot, {
      foreignKey: "tier_id"
    });
  });

  it("should associate tier with message", () => {
    const Tier = tier(this.sequelize, this.dataTypes);
    Tier.associate(this.models);

    expect(Tier.hasMany).toHaveBeenCalledWith(this.models.Message, {
      foreignKey: "tier_id"
    });
  });
});
