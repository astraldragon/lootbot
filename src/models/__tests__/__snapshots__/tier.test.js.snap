// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/tier should create the model definition 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Tier",
      Object {
        "color": "integer",
        "guild": "string",
        "image": "string",
        "luckyWeight": "float",
        "name": "string",
        "weight": "float",
      },
      Object {},
    ],
  ],
}
`;
